scIB/metrics.py

```diff
@@ -710,14 +710,24 @@ def pc_regression(data, variable, pca_var=None, n_comps=50, svd_solver='arpack',
     else:
         X_pca = matrix
         n_comps = matrix.shape[1]
-    
+
     ## PC Regression
     if verbose:
         print("fit regression on PCs")
-    
-    # one-hot encode categorical values
-    variable = pd.get_dummies(variable).to_numpy()
-    
+
+    # handle categorical values
+    try:
+        categorical = variable.dtype.name == 'category'
+    except:
+        categorical = not isinstance(variable[0], (int, float))
```

Comment:
Why use error handling and not if else?

Reply:
You mean check if the variable is of type series/dataframe with  an if/else? I found  this approach to be more readable and general towards whether it congestions a dtype or not.  Would you prefer if/else type checking instead?

Reply:
yes. I don't think we should catch all errors like this when we know what we are looking for. Ideally check if categorical, check if numerical... if neither throw an error.

Reply:
Please do this as if else...

Reply:
I included a different  type check (with if/else) which will be in the  next commit, once the values of kBet are reproducible